3rd state to Scientist's Serendipity.

Eneocho opened the pull request:
Scientist's Serendipity felt a bit underwhelming when compared to it's supposed effect, since you would already have a decent chunk of the research it provides (if not all) by the time you make some. Therefore, if you make/acquire 50u of the stuff on a drinking glass, you get Scientist's Sapience, which maxes all research to their limits, except for Anomaly, which isn't affected (so i don't fuck up xenoarch) and Illegal Tech, which only gets to level 2 (Enough to give you a headstart).

Serendipity has Materials 7, Engineering 3, Plasma 2, Power 4, Bluespace 6, Combat 3, Electromagnetic 6 and Data Theory 3.
Sapience has Materials 10, Engineering 5, Plasma 4, Power 5, Bluespace 10, Medical 5, Combat 6, Electromagnetic 6, Data Theory 5, NT Experimental 1 and Illegal Tech 2.

:cl:
 * rscadd: Added another variant for Scientist's Serendipity, which maxes all research, except Anomaly, which isn't affected and Illegal, which only gets to level 2.

Eneocho commented:
>code/game/centcomm_orders/orders_service.dm, line 88, column 6:
>error: failed to resolve path /datum/reagent/vinegar

What? Should I commit something random so it updates?

gurfan commented:
your indentation is messed up @Eneocho

Exxion commented on code/modules/reagents/Chemistry-Reagents.dm:
```diff
@@ -6302,16 +6302,22 @@
-		D.origin_tech = "materials=7;engineering=3;plasmatech=2;powerstorage=4;bluespace=6;combat=3;magnets=6;programming=3"
-
+		switch(volume)
+			if(1 to 9)
```
`volume` isn't an integer. This should be `0 to 10` and the other one should also go all the way up to the bottom of the next. Otherwise nothing would happen if it were between 9 and 10 units or between 49 and 50 units.
Oh and the order should probably be reversed so you hit it if you get the minimum instead of just above the minimum

Eneocho replied:
>This should be 0 to 10 and the other one should also go all the way up to the bottom of the next. Otherwise nothing would happen if it were between 9 and 10 units or between 49 and 50 units.

So it should be if(0 to 10), if(10 to 50) and if (50 to INFINITY)?

>Oh and the order should probably be reversed so you hit it if you get the minimum instead of just above the minimum

As in putting the if conditions in descending order or putting it like if(50 to 10)?

Eneocho replied:
Also, the original Serendipity code had "if(volume <10)" as it's volume check. I wanted to use it but didn't know how (can you put 2 conditions on that? as in "if(volume >10,<50)?), so I went with "if(10 to 50)".

Exxion replied:
For your first question, yes. For your second question, no, reversed as in put the `50 to INFINITY` one first and the `0 to 10` one last.
Yes, you can do `(volume > 10) && (volume < 50)`

Eneocho replied:
Is this better?

Eneocho commented:
> 
> 
> your indentation is messed up @Eneocho

I copied the volume part from Dan soda's code.

Eneocho commented:
No idea how to fix that. I didn't even touch the vinegar reagent nor centcomm's orders.

Exxion commented on code/modules/reagents/Chemistry-Reagents.dm:
```diff
@@ -6302,16 +6302,23 @@
 		glass_desc = "There is as yet insufficient data for a meaningful answer."
-	else
+
+	if(volume > 10) && (volume < 50)
```
The `(volume > 10) && (volume < 50)` would have to be wrapped in a pair of parentheses, but I would just recommend doing
```suggestion
	else if(volume < 50)
```
instead.

Exxion commented on code/modules/reagents/Chemistry-Reagents.dm:
```diff
@@ -6302,16 +6302,23 @@
 
+	else
+		glass_icon_state = "scientists_serendipity"
+		glass_name = \improper Scientist's Lucky Datadisk"
```
Oh, here's the problem.
```suggestion
		glass_name = "\improper Scientist's Lucky Datadisk"
```

Eneocho replied:
Thanks for noticing that.

Kurfursten commented on code/modules/reagents/Chemistry-Reagents.dm:
```diff
@@ -6302,16 +6302,23 @@
 		glass_name = "\improper Scientist's Surprise"
 		glass_desc = "There is as yet insufficient data for a meaningful answer."
-	else
+
```
needs to blank origin tech when it goes back down so that you can't just fill it up to 50 once and then empty it out to 1u

Eneocho replied:
It's how it was coded originally, the surprise had no origin tech, while the serendipity had. Should I add this?
>D.origin_tech = ""

Kurfursten replied:
That will do it, yeah. This is a good chance to fix the original oversight.

Eneocho replied:
Fixed.

Kurfursten commented on code/modules/reagents/Chemistry-Reagents.dm:
```diff
@@ -6302,16 +6302,23 @@
 
+	else
+		glass_icon_state = "scientists_serendipity"
+		glass_name = "\improper Scientist's Lucky Datadisk"
```
This name doesn't match the naming convention of alliteration
> Doctor's Delight
> Scientist's Surprise
> Scientist's Serendipity

Eneocho replied:
Scientist's Sapience it is now.

Kurfursten reviewed (changes requested):
those things I said

Eneocho commented:
Changed the name to Scientist's Sapience (wisdom) to go along with both the theme and aliteration. 
Due to feedback on the thread, added NT tech level 1 and Syndicate level 2.

Kurfursten commented:
I thought it was better not giving illegal/NT. I also don't think the NT research will even work since it's a hidden tech, so you need to test that.

Sooose commented:
Can't wait to abuse this.

Kurfursten reviewed (approved):
code is fine

GreatBigFailure commented:
It's so rarely used I'd prefer it just max everything and call it a day. I like the addition either way.
